core/libcamera_app.cpp

```diff
@@ -548,15 +545,20 @@ void LibcameraApp::StartCamera()
 	if (!controls_.get(controls::Sharpness))
 		controls_.set(controls::Sharpness, options_->sharpness);
 
-	if (!controls_.get(controls::AfMode))
-		controls_.set(controls::AfMode, options_->afMode_index);
-	if (!controls_.get(controls::AfRange))
-		controls_.set(controls::AfRange, options_->afRange_index);
-	if (!controls_.get(controls::AfSpeed))
-		controls_.set(controls::AfSpeed, options_->afSpeed_index);
-	if (!controls_.get(controls::LensPosition))
-		controls_.set(controls::LensPosition, options_->lens_position);
-
+	if(camera_->controls().count(&controls::AfMode)>0){
+		LOG(2, "Camera has AfMode");
+		if (options_->afMode_index!=-1 && !controls_.get(controls::AfMode)){
+			controls_.set(controls::AfMode, options_->afMode_index);
+			if(options_->afMode_index == controls::AfModeAuto){
+				//does the first focus on start 
+				controls_.set(controls::AfTrigger, controls::AfTriggerStart);
+			}
+		}
+		if (options_->afRange_index!=-1 && !controls_.get(controls::AfRange))
+			controls_.set(controls::AfRange, options_->afRange_index);
+		if (options_->afSpeed_index!=-1 && !controls_.get(controls::AfSpeed))
+			controls_.set(controls::AfSpeed, options_->afSpeed_index);
+	}
```

Comment:
Thanks for adding `AfTrigger`, but I think it should not be in `StartCamera()`. The problem is that it will re-trigger when capturing a still, without waiting for completion. Depending on the implementation, it might result in blurry image.

We could make it conditional on first start or on stream type... but rather than trying to 'guess' the application's intention, it would be cleaner to deal with it at the individual application level.

Could you please remove `AfTrigger` for now. It will be re-worked in a later commit.